app/controllers/users_controller.rb

```diff
@@ -0,0 +1,28 @@
+class UsersController < ApplicationController
+  allow_unauthenticated_access only: %i[show]
+  before_action :resume_session, only: %i[show]
+  before_action :set_user, only: %i[show toggle_status]
+  before_action :check_inactive_user, only: %i[show]
+  before_action :require_admin, only: %i[ toggle_status ]
+
+  def show; end
+
+  def toggle_status
+    @user.toggle_status!
+    redirect_to user_path(@user), notice: t(".success")
+  end
+
+  private
+
+  def set_user
+    @user = User.unscoped.find(params[:id])
+  end
+
+  def check_inactive_user
+    redirect_to root_path if @user.inactive? && (Current.user&.regular? || !Current.user)
```

Comment:
And here comes the duplication. So, why don't we create a method on `ApplicationController` or even a `Concern` to handle this?